### PR TITLE
Remove old ignored tests

### PR DESCRIPTION
**Pull request by steveklabnik**
Two commits here: one which removes a bunch of tests, and re-enables a few that work.

Second updates the syntax of one of the failing tests. It still doesn't pass, but at least it compiles.

### CODE REVIEW

**Comment by tshepang** [comment]
wow, that's a lot of shed weight

File: src/test/compile-fail/dead-code-closure-bang.rs [hunk 1]
```diff
@@ -15,5 +15,5 @@
 fn main() {
     let x = || panic!();
     x();
-    std::io::println("Foo bar"); //~ ERROR: unreachable statement
+    println!("Foo bar"); //~ ERROR: unreachable statement
```
Comment (Manishearth) [inline src/test/compile-fail/dead-code-closure-bang.rs]: This test is still ignored?

Comment (steveklabnik) [inline src/test/compile-fail/dead-code-closure-bang.rs]: yes, it is not fixed, I just updated the syntax.

**Pull request merged**